### PR TITLE
Keep attribute on the same line as a member if they started on a single line

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/11572

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
retest vsi please

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

**Comment by CyrusNajmabadi** [comment]
retest windows_debug_unit64_prtest please

**Comment by CyrusNajmabadi** [comment]
retest windows_vsi_p2_open_prtest please

**Comment by CyrusNajmabadi** [comment]
retest windows_vsi_p2_open_prtest  please

File: src/Workspaces/CSharp/Portable/Formatting/Rules/SuppressFormattingRule.cs [hunk 1]
```diff
@@ -64,8 +65,24 @@ private void AddSpecificNodesSuppressOperations(List<SuppressOperation> list, Sy
             var memberDeclNode = node as MemberDeclarationSyntax;
             if (memberDeclNode != null)
             {
+                // Attempt to keep the part of a member that follows hte attributes on a single
```
Comment (Pilchie) [inline src/Workspaces/CSharp/Portable/Formatting/Rules/SuppressFormattingRule.cs]: `hte`

**Comment by Pilchie** [comment]
This has been a longstanding request, and I think we should support it.  However, I'm a little wary of changing the formatting engine unilaterally these days.  I wonder if we should add a `Wrapping` option for this.  Thoughts @DustinCampbell?

**Comment by Pilchie** [comment]
:+1: on the code change itself though.

**Comment by CyrusNajmabadi** [comment]
Note: this is not a unilateral approach.  Both wrap/suppress operations are predicated on the attribute being on the same line in the first place.  If you have your member spread over multiple lines, we'll respect that.

**Comment by CyrusNajmabadi** [comment]
retest  windows_vsi_p2_open_prtest  please

**Comment by jmarolf** [comment]
:+1:

**Comment by Pilchie** [comment]
@CyrusNajmabadi I get that it's only if they start on the same line, but my supposition is that there may be people who like that formatting "fixes" their code when they are in that state.

**Comment by CyrusNajmabadi** [comment]
Good point. @DustinCampbell What do you think?  Does this warrant an option or not?

Note: right now users can get the formatting they want, regardless of which style the prefer.  If they prefer attributes formatted on the same line, then that will be preserved if there is an enter.  If they like it on the same line, then that will be preserved if the member is all on one line.

All that changes is what happens by default when a single-line member with attributes is formatted.  Now it will stay on one line instead of auto moving to two lines.

Personally, i think that's an ok change in behavior.  But i'm ok adding an option if people think that's necessary.

**Comment by DustinCampbell** [comment]
I think this is fine. So long as we're not moving the attribute from where the user places it.

**Pull request merged**